ARCHITECTURE.md

```diff
@@ -0,0 +1,75 @@
+# Architecture
+
+> [!NOTE]
+> This document is intended to be prescriptive, and any deviation from
+> implementation should be treated as a bug in the implementation. That is not
+> to say that this document will not contain any errors. Please use your best
+> judgement, when encountering any said deviations!
+
+This document is intended to provide an architectural overview of the
+single-page applications that this repository produces. The goal is to
+familiarize the reader with the code base, and how various packages and
+dependencies come together to in a final product.
+
+Efforts will be made to be general, providing specific examples as warranted
+Generally speaking, specific examples should be avoided when possible, since
+they can deviate from what is described in this document.
+
+## Single-page Application
+
+This repository produces the javascript and css code that is use by all
+VEuPathDB websites: genomics-site, clinepi-site, mbio-site, and ortho-site. All
+of these sites follow the same architecture, which is based on the React
+library.
+
+In order to use the code in a website, the top-level javascript file and css
+file must be included on a webpage. Because the application is a single-page
+application, routing is done client-side. It is standard practice to service the
+html that for the webpage in such a way that requests for URLs that start with a
+specific path are all routed to the HTML file. The javascript code will
+dynamically determine what to display on the screen, based on the URL.
+
+For example, you might have an Apache mod_rewrite rule that looks like this:
```

Comment:
perhaps because I am not familiar with Apache mod_rewrite, but I lost you here.

Reply:
I tried to clarify. Take a look at https://github.com/VEuPathDB/web-monorepo/blob/c44e50dae0c64e0719dce0aecbc95797cf61910b/ARCHITECTURE.md?plain=1#L40-L46